src/coreclr-debug/activate.ts

```diff
@@ -113,3 +110,46 @@ function showDotnetToolsWarning(message: string) : void
             });
     }
 }
+
+interface AdapterExecutableCommand {
+    command: string;
+}
+
+// The default extension manifest calls this command as the adapterExecutableCommand
+// If the debugger components have not finished downloading, the proxy displays an error message to the user
+// Else it will launch the debug adapter
+export async function registerAdapterExecutionCommand(channel: vscode.OutputChannel): Promise<AdapterExecutableCommand> {
+    let logger = new Logger(text => channel.append(text));
+    let util = new CoreClrDebugUtil(common.getExtensionPath(), logger);
+
+    // Check for .debugger folder. Handle if it does not exist.
+    if (!CoreClrDebugUtil.existsSync(util.debugAdapterDir()))
+    {
+        // our install.complete file does not exist yet, meaning we have not completed the installation. Try to figure out what if anything the package manager is doing
+        // the order in which files are dealt with is this:
+        // 1. install.Begin is created
+        // 2. install.Lock is created
+        // 3. install.Begin is deleted
+        // 4. install.complete is created
+
+        // install.Lock does not exist, need to wait for packages to finish downloading.
+        let installLock: boolean = await common.installFileExists(common.InstallFileType.Lock);
+        if (!installLock) {
+            throw new Error('The omnisharp-csharp extension is still downloading packages. Please see progress in the output window below.');
```

Comment:
Show that output channel?

Reply:
Oh, and extension is called 'C#' in the extensions gallery and also that is the name of the output channel, so I would probably suggest using 'C#' instead of 'omnisharp-csharp' in the two error messages.